octodns/record/__init__.py

```diff
@@ -429,6 +429,10 @@ def __init__(self, _id, data):
         ]
         values.sort(key=lambda d: d['value'])
 
+        # normalize weight of a single-value pool
+        if len(values) == 1:
+            values[0]['weight'] = 1
```

Comment:
Having a hard time figuring out the path that would lead to this being required. https://github.com/octodns/octodns/pull/706/files#diff-7f4d8ad4d1143b84eb1f75b328f01cf688b9421abc8d0295077ac94922814020R427 in the list comprehension for `data['values']` above seems like it should set the weight to `1` when it's not define. Is this trying to "fix" when people put a non-`1` weight on a single value? If so 👍, but we should probably stick a `logger.warn` here telling them that it's happening so they'll have a chance to learn their config doesn't make sense.

Reply:
> "fix" when people put a non-1 weight on a single value?

yep exactly. will add the `logger.warn`.

Reply:
Added the warning.

Reply:
Sorry it didn't occur to me when I was looking at this and just popped into my head this afternoon when I was AFK. I believe this weight check should actually be a validation rather than an auto-fix & warning as many things of this sort are already pointed out at that phase and people can always configure `lenient` if they have a strong reason to preserve something that doesn't agree.

I've PR'd that over in https://github.com/octodns/octodns/pull/715 and will pull this and the test change below out of this one in favor of that, and to avoid requiring another round trip of suggestion, change, review.